decidim-core/app/packs/src/decidim/append_redirect_url_to_modals.js

```diff
@@ -53,23 +53,31 @@ $(() => {
   }
 
   $(document).on("click.zf.trigger", (event) => {
-    const target = `#${$(event.target).data("dialogOpen")}`;
-    const redirectUrl = $(event.target).data("redirectUrl");
-
-    if (!target || !redirectUrl) {
-      return;
+    // Try to get the <a> directly or find the closest parent <a>
+    let $target = $(event.target);
+    if (!$target.is("a")) {
+      // Find the closest parent <a> if the click is not directly on an <a>
+      $target = $target.closest("a"); 
     }
 
-    $("<input type='hidden' />").
-      attr("id", "redirect_url").
-      attr("name", "redirect_url").
-      attr("value", redirectUrl).
-      appendTo(`${target} form`);
+    // Check if an <a> was found
+    if ($target.length) {
+      const dialogTarget = `#${$target.data("dialog-open")}`;
+      const redirectUrl = $target.data("redirectUrl");
```

Comment:
@rsdeus I suggest to use if/else here to return early more than drawing a "golden path". 

```suggestion
    // Check if an <a> was found
    if (!$target)
      return;
      
    const dialogTarget = `#${$target.data("dialog-open")}`;
    const redirectUrl = $target.data("redirectUrl");
    if(!dialogTarget || !redirectUrl)
      return;
      
    // preconditions are OK, you can then do the rest
```

Reply:
Done! Thanks :)